server.js

```diff
@@ -1190,7 +1186,7 @@ app.get('/get-task/:version(\\d+)', asyncMiddleware( async (req, res, next) => {
             });
         }
 
-        if (pending_matches[pending_matches.length - 1].game_count % 2) {
+        if (match.game_count % 2) {
```

Comment:
This would send out the same color multiple times in a row until a match result came back and we incremented game_count, if we even do that elsewhere.

Should it be match.game_count + match.requests.length so it'll alternate as each match goes out?

Reply:
Oh wait you still increment game_count? Never mind then, I thought you were removing that.

Reply:
I was going to remove game_count but then noticed it was used for sending black/white so i kept it in :)

Reply:
If you remove it later just remember to increment it when we have a match come in, in your loop that does the filter/foreach. (Remove request, increment game_count.) And then you'd do black/white on game_count + requests.length.

Do you think it's cleaner to do it that way?

Reply:
I would use a new variable for the purpose of sending black/white matches. That one would only toggle 0 and 1.

Reply:
Lets maybe make this  `if (match.game_count + match.requests.length) % 2`  and  increment game_count when we lower the requests when a match arrives.